lightning/src/util/errors.rs

```diff
@@ -47,6 +49,15 @@ pub enum APIError {
 	/// An attempt to call watch/update_channel returned an Err (ie you did this!), causing the
 	/// attempted action to fail.
 	MonitorUpdateFailed,
+	/// [`KeysInterface::get_shutdown_scriptpubkey`] returned a shutdown scriptpubkey incompatible
+	/// with the channel counterparty as negotiated in [`InitFeatures`].
```

Comment:
This should mention what the user should do. eg "Sticking to SegWit V0 scripts may resolve this issue. If you cannot, you will not be able to open a channel or cooperatively close one with this peer (and will have to force-close instead)."